core/src/main/java/org/axonframework/queryhandling/AnnotationQueryHandlerAdapter.java

```diff
@@ -0,0 +1,59 @@
+package org.axonframework.queryhandling;
+
+import org.axonframework.common.Registration;
+import org.axonframework.messaging.annotation.ParameterResolver;
+import org.axonframework.messaging.annotation.ParameterResolverFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+
+/**
+ * @author Marc Gathier
+ * @since 3.1
+ */
+public class AnnotationQueryHandlerAdapter implements QueryHandlerAdapter {
+    private final Object target;
+    private final ParameterResolverFactory parameterResolverFactory;
+
+    public AnnotationQueryHandlerAdapter(Object target, ParameterResolverFactory parameterResolverFactory) {
+        this.target = target;
+        this.parameterResolverFactory = parameterResolverFactory;
+    }
+
+    @Override
+    public Registration subscribe(QueryBus queryBus) {
+        Arrays.stream(target.getClass().getMethods()).forEach(m -> {
+            if( m.isAnnotationPresent(QueryHandler.class)) {
+                QueryHandler qh = m.getAnnotation(QueryHandler.class);
+                String queryName = qh.queryName().isEmpty() ? m.getParameters()[0].getType().getName() : qh.queryName();
+                String responseName = qh.responseName().isEmpty() ? m.getReturnType().getName() : qh.responseName();
+                ParameterResolver[] parameterResolvers = new ParameterResolver[m.getParameterCount()];
+                for( int i = 0 ; i < m.getParameterCount(); i++) {
+                    parameterResolvers[i] = parameterResolverFactory.createInstance(m, m.getParameters(), i);
```

Comment:
Typically if a `parameterResolverFactory.createInstance()` cannot be created for the given parameter, the `ParameterResolverFactory` returns `null`.
So I believe you'd thus have nulls in the `ParameterResolver[] parameterResolvers`, which might lead to issues in the `runQuery()` function on line 46.

Reply:
I believe the `AnnotatedMessageHandlingMember` it's constructor tries to solve this, maybe we can reuse that bit of logic.